cosmicfishpie/cosmology/cosmology.py

```diff
@@ -753,9 +888,252 @@ def f_cb_deriv(k_array, k_fix=False, fixed_k=1e-3):
+        - Debug information is printed if the upr.debug flag is set.
+        """
+        upr.debug_print("DEBUG: cosmopars = ", cosmopars)
+        symbpars = deepcopy(cosmopars)
+        if "h" in symbpars:
+            symbpars["h"] = symbpars.pop("h")
+        if "H0" in symbpars:
+            H0 = symbpars.pop("H0")
+            symbpars["h"] = H0 / 100.0
+        if "ombh2" in symbpars:
+            symbpars["Omegab"] = symbpars.pop("ombh2") / (symbpars["h"] ** 2)
+        if "omch2" in symbpars:
+            symbpars["Omegam"] = (symbpars.pop("omch2") / symbpars["h"] ** 2) + symbpars["Omegab"]
+            # Omegam = Omegac + Omegab
+        # ["sigma8", "As", "logAs", "10^9As", "ln_A_s_1e10"]
+        try:
+            if "As" in symbpars:
+                symbpars["10^9As"] = 10**9 * symbpars.pop("As")
+            if "logAs" in symbpars:
+                symbpars["10^9As"] = 10**9 * (np.exp(symbpars.pop("logAs")) * 1.0e-10)
+            try:
+                As_value = symbpars.get("10^9As")
+                upr.debug_print("DEBUG: symbpars['10^9As'] = ", As_value)
+                if np.isscalar(As_value):
+                    try:
+                        symbpars["sigma8"] = self.symblin.As_to_sigma8(
+                            symbpars["10^9As"],
+                            symbpars["Omegam"],
+                            symbpars["Omegab"],
+                            symbpars["h"],
+                            symbpars["ns"],
+                        )
+                        upr.debug_print("DEBUG: symbpars['sigma8'] = ", symbpars["sigma8"])
+                    except:
+                        print("As to sigma8 conversion failed")
+                        raise ValueError
+            except KeyError:
+                pass
+            try:
+                sigma8_value = symbpars.get("sigma8")
+                if np.isscalar(sigma8_value):
+                    try:
+                        As_n = self.symblin.sigma8_to_As(
+                            symbpars["sigma8"],
+                            symbpars["Omegam"],
+                            symbpars["Omegab"],
+                            symbpars["h"],
+                            symbpars["ns"],
+                        )
+                        upr.debug_print("DEBUG: As_n = ", As_n)
+                        symbpars["10^9As"] = As_n
+                    except:
+                        print("sigma8 to As conversion failed")
+                        raise ValueError
+                else:
+                    print("sigma8 value not scalar")
+                    upr.debug_print("DEBUG: symbpars = ", symbpars)
+            except KeyError:
+                print("sigma8 or 10^9As not in symbpars")
+                raise KeyError
+        except:
+            print("sigma8<->As conversion failed for other reasons")
+            upr.debug_print("DEBUG: symbpars = ", symbpars)
+            raise ValueError
+        return symbpars
```

Comment:
I am not sure about this. The symbo emulator was trained using some specific tau, mnu and so on. Which we will never Fisher over. But it's good to keep them for the user to know what to compare against. You think it is safer to pop them to avoid someone fishering them?